pkg/trace/writer/trace_test.go

```diff
@@ -53,6 +54,60 @@ func TestTraceWriter(t *testing.T) {
 	})
 }
 
+func TestTraceWriterMultipleEndpointsConcurrent(t *testing.T) {
+	var (
+		srv = newTestServer()
+		cfg = &config.AgentConfig{
+			Hostname:   testHostname,
+			DefaultEnv: testEnv,
+			Endpoints: []*config.Endpoint{
+				{
+					APIKey: "123",
+					Host:   srv.URL,
+				},
+				{
+					APIKey: "123",
+					Host:   srv.URL,
+				},
+			},
+			TraceWriter: &config.WriterConfig{ConnectionLimit: 200, QueueSize: 40},
+		}
+		numWorkers      = 100
+		numOpsPerWorker = 100
+	)
+
+	t.Run("ok", func(t *testing.T) {
```

Comment:
We can drop this line and the wrapping. It's just one test, no need for a sub-test.